iot-hub/Tutorials/ConnectivityTests/SimulatedDevice-2.js

```diff
@@ -14,28 +14,42 @@ var Message = require('azure-iot-device').Message;
 var NoRetry = require('azure-iot-common').NoRetry;
 var crypto = require('crypto');
 
-// Example code showing how to generate a SAS token.
-var generateSasToken = function(resourceUri, signingKey, policyName, expiresInMins) {
-  resourceUri = encodeURIComponent(resourceUri);
+// // Example code showing how to generate a SAS token using the SDK.
+// function generateSasToken(resourceUri, signingKey, policyName, expiresInMins) {
+//   resourceUri = encodeURIComponent(resourceUri);
 
-  // Set expiration in seconds
-  var expires = (Date.now() / 1000) + expiresInMins * 60;
-  expires = Math.ceil(expires);
-  var toSign = resourceUri + '\n' + expires;
+//   // Set expiration in seconds
+//   var expires = (Date.now() / 1000) + expiresInMins * 60;
+//   expires = Math.ceil(expires);
+//   var toSign = resourceUri + '\n' + expires;
 
-  // Use crypto
-  var hmac = crypto.createHmac('sha256', new Buffer(signingKey, 'base64'));
-  hmac.update(toSign);
-  var base64UriEncoded = encodeURIComponent(hmac.digest('base64'));
+//   // Use crypto
+//   var hmac = crypto.createHmac('sha256', new Buffer(signingKey, 'base64'));
+//   hmac.update(toSign);
+//   var base64UriEncoded = encodeURIComponent(hmac.digest('base64'));
 
-  // Construct autorization string
-  var token = "SharedAccessSignature sr=" + resourceUri + "&sig="
-  + base64UriEncoded + "&se=" + expires;
-  if (policyName) token += "&skn="+policyName;
-  return token;
-};
-
-// There are two ways to create a SAS token for a device to connect:
+//   // Construct authorization string
+//   var token = "SharedAccessSignature sr=" + resourceUri + "&sig="
+//   + base64UriEncoded + "&se=" + expires;
+//   if (policyName) token += "&skn="+policyName;
+//   return token;
+// };
+//
+// // Example code showing how to generate a SAS token using the SDK
+// 
+// var SharedAccessSignature = require('azure-iot-common').SharedAccessSignature;
+// function generateSasToken(resourceUri, signingKey, policyName, expiresInMins) {
+//
+//   // Set expiration in seconds
+//   var expires = (Date.now() / 1000) + expiresInMins * 60;
+//   expires = Math.ceil(expires);
+//
+//   var token = SharedAccessSignature.create(resourceUri, policyName, signingKey, expires);
+//
+//   return token;
+// };
+//
+// // There are two ways to create a SAS token for a device to connect:
 //
 // var token = generateSasToken(
 //  'hubname.azure-devices.net/devices/devicename',
```

Comment:
are both of these meant to be commented out? what's the choice or action that a beginner user should take?

Reply:
This commented code is referenced from the tutorial article.